crypto/jwks/component.go

```diff
@@ -94,11 +100,20 @@ func (k *jwksCrypto) Features() []contribCrypto.Feature {
 }
 
 // Init the JWKS object from the metadata property
-func (k *jwksCrypto) initJWKS(ctx context.Context) error {
+func (k *jwksCrypto) initJWKS() error {
 	if len(k.md.JWKS) == 0 {
 		return errors.New("metadata property 'jwks' is required")
 	}
 
+	// Use context based on close channel.
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Shouldn't close also be based on incoming context?

Reply:
I think so, but this is what we have currently so left it.

Reply:
Whoops, that's not true at all. Sorry.

Reply:
Fixed

Reply:
I have reverted that change because we don't want the context by the `cache`, `fswatcher` etc. to be the same as the `Init` context. If it was and the runtime caller adds a timeout, for example, it will cancel the processes. By using a Background, the runtime can control the process shutdown with `Close` instead.